ADR/0008-integration-service-api-contracts.md

```diff
@@ -0,0 +1,266 @@
+# 8. HACBS Test Stream - API contracts
+
+Date: 2022-11-28
+
+## Status
+
+Accepted
+
+## Context
+
+The Hybrid Application Cloud Build Service being developed aims to serve Red Hat teams but also partners and customers. This requires a level of adaptability to avoid recreating custom flows and Tasks for each stakeholder.
+
+In this respect Tasks developed by HACBS test stream should allow swapping external systems to accommodate different environments. This swap should not induce the complete recreation of pipelines.
+
+This and the idea of providing a homogeneous experience, which is easier to comprehend and navigate complex systems, leads to the definition of `API contracts`. These contracts need to be understood as guidance that may evolve with time and experience while keeping the aim of building a flexible homogeneous system.
+
+Tasks from the test stream may exchange information through
+
+1. Parameters passed at the time of the trigger of a PipelineRun (that is user, git repository, etc)
+2. Output of upstream elements of the Pipeline which may be needed for performing the Task (that is build artifacts, container image, etc)
+3. Providing whether the Task was successful or not
+4. Output of the Task that may be needed for processing subsequent Tasks of the pipeline
+5. Output of the Task that need to be stored for auditing and troubleshooting (that is summarized results of a validation or a scan, etc)
+6. Communication with external systems that are needed for performing the Task (that is container image repository, image scanner, etc)
+7. Connection details and credentials that may be required for the above
+
+## Decision
+
+### Results and Records
+
+Related to [PLNSRVCE-41](https://issues.redhat.com/browse/PLNSRVCE-41) investigations in regards to [Tekton results](https://github.com/tektoncd/results). The investigations are documented [here](https://docs.google.com/document/d/1WGWx6MllVs-BwRLd0PP5QTQHgGp0fW2dOYRKRAKWiwo/edit#heading=h.bpyxfhyud4dv).
+
+The output of each Tekton task will be provided in two forms:
+- As a minimized [Tekton result](https://tekton.dev/docs/pipelines/tasks/#emitting-results) in JSON format listing all test failures
+  - The name of the result will be **HACBS_TEST_OUTPUT**
+- As a Test output JSON file with full test information that’s saved in the Tekton Pipeline Workspace
+  - The name of the file will be in snake case and will be in the form of **test_name_output.json**
+
+#### Tekton Results
+
+The Test output of the Tekton result will be a JSON object that includes the context about the test along with the list of check names for all failed checks
+
+The output will provide the following information about the overall test result:
```

Comment:
Do we happen to have this format encoded as [json schema](https://json-schema.org/) anywhere? That could be nice to link to or copy here directly if we have it. Nicely unambiguous.

Reply:
I don't think we have json schema for it, but we generate it with `jq`. @dirgim Could you help confirm it?

Reply:
All good - just curious. Thanks!